math/src/fft/helpers.rs

```diff
@@ -0,0 +1,10 @@
+use super::errors::FFTError;
+
+pub fn log2(n: usize) -> Result<u64, FFTError> {
+    if !n.is_power_of_two() {
+        return Err(FFTError::InvalidOrder(
+            "The order of polynomial should a be power of 2".to_string(),
+        ));
+    }
```

Comment:
Is rounding down not an option here?

Reply:
Could you elaborate? a hypothesis of the implemented FFT algorithm is that the amount of coefficients needs to be a power of two

Reply:
Ok, so that requirement is specific to fft. I thought of log2 in general where most implementations just round down (i.e. return the index of the most significant 1 bit).